mycroft/client/speech/mic.py

```diff
@@ -303,11 +305,22 @@ def _wait_until_wake_word(self, source, sec_per_buffer):
         counter = 0
 
         while not said_wake_word:
-            if check_for_signal('buttonPress'):
-                logger.debug("Button Pressed, wakeword not needed")
+            # Check if told programatically to skip the wake word for dialog
+            if check_for_signal('startListening'):
                 said_wake_word = True
                 continue
 
+            # Pressing the button can start recording (unless it is being
+            # used to mean 'stop' instead)
+            if check_for_signal('buttonPress', 1):
+                # give others time to consume this as 'stop'
+                sleep(0.25)
+                if check_for_signal('buttonPress'):
+                    # signal is still here, assume it was to begin recording
+                    logger.debug("Button Pressed, wakeword not needed")
+                    said_wake_word = True
+                    continue
+
```

Comment:
There is an architectural change needed.  Basically, the way "stop" is implemented using the message bus, we don't now if a Skill consumed the button press.  The mechanism I tossed in there using the 1/4 second sleep allows me to detect if the mimic_tts process was active (i.e. if the unit was talking) and it consumes the buttonPress to stop speaking and viseme playback.  But there is no way to detect other consumptions, e.g. if NPR news is playing and the button press is just meant to stop that.

I might do something hacky for short-term such for anything that actually would need to "stop" independent of speaking -- such as NPR news or music playback.  I can have those call a function that consumes the signal.  I'll probably have them call a MycroftSkill.do_stop() helper function.

In the longer term, we need a SkillManager that is aware of the Skill processes that we have launched.  It will invoke the method on those Skills and that method will return either True or False depending on whether they stopped.  Then the SkillManager will be the one that consumes the buttonPress.

Make sense?